backend/baseclass.pm

```diff
@@ -657,12 +657,14 @@ sub select_console ($self, $args) {
     my $testapi_console = $args->{testapi_console};
 
     my $selected_console = $self->console($testapi_console);
-    my $activated = try {
-        local $SIG{__DIE__} = 'DEFAULT';
-        $selected_console->select;
-    }
-    catch {
-        {error => $_};
+    my $activated = do {
+        try {
+            local $SIG{__DIE__} = 'DEFAULT';
+            $selected_console->select;
+        }
+        catch ($e) {
+            {error => $e};
+        }
```

Comment:
It looks like this could be simplified by replacing the
```
return $activated if ref($activated);
```
below with this:
```suggestion
        catch ($e) { return {error => $e} }
```
The subtle difference between implicitly returning from the block and explicitly `return`ing from the sub might be hard to get at first.
But this case is an example where it comes in handy.